src/util/NumberRank.js

```diff
@@ -0,0 +1,29 @@
+var ordinal = require('ordinal').english
+
+var NumberRank = function(value, unsortedValues, ascending) {
+  var sortedValues = unsortedValues.sort(function(a, b) { return b - a})
+  if(ascending) sortedValues.reverse()
+
+  var indexOfValue = sortedValues.indexOf(value)
+  var rankArray = []
+  for(var i = 0; i < sortedValues.length; i ++) {
+    if(sortedValues[i-1]) {
+      // If the previous value equals the current value, the current value is tied/set to the same rank as the previous value
+      if(sortedValues[i-1] === sortedValues[i]) {
+        rankArray.push(rankArray[i-1])
+      }
+      else {
+        rankArray.push(i+1)
+      }
+    }
+    else { // Start with rank 1 for the first element
+      rankArray.push(1)
```

Comment:
There's a trick you can do here to reduce nesting. Before the for loop, add `rankArray.push(1)` and in the for loop, it should look like `for(var i = 0; i < sortedValues.length; i ++)`. I'm actually surprised that line 10 doesn't break.